File: element_is_index.rb
#!/usr/bin/env ruby
# http://programmingpraxis.com/2013/07/26/find-xi-i-in-an-array/

element_is_i = -> list { list.each_index {|i| return i if list[i] == i } }

p element_is_i[[-3,0,1,3,5,7]]

File: element_is_index.swift
#!/usr/bin/env swift
// http://programmingpraxis.com/2013/07/26/find-xi-i-in-an-array/

func elementIsIndex(list: [Int]) -> Int? {
  for var i = 0; i < list.count; i++ {
    if list[i] == i {
      return i
    }
  }
  return nil
}

let elements = [-3,0,1,3,5,7]

println(elementIsIndex(elements))